bodhi/tests/server/consumers/test_masher.py

```diff
@@ -1588,3 +1749,182 @@ def test_urlerror(self, urlopen, sleep, publish):
         urlopen.assert_has_calls(expected_calls)
         t.log.exception.assert_called_once_with('Error fetching repomd.xml')
         sleep.assert_called_once_with(200)
+
+
+class TestPungiMashThread(object):
+
+    @mock.patch("bodhi.server.consumers.masher.Compose")
+    @mock.patch("pungi.notifier.PungiNotifier")
+    @mock.patch("bodhi.server.consumers.masher.PungiWrapper")
+    def test_run(self, pungi_wrapper, *args):
+        compose_id = "f27-modular-updates"
+        pungi_conf = mock.Mock()
+        target_dir = "/tmp"
+        variants_conf = mock.Mock()
+        logger = mock.Mock()
+        pungi_wrapper.init_compose_dir = mock.Mock()
+        mash = PungiMashThread(compose_id, target_dir, pungi_conf, variants_conf, logger)
+
+        assert not mash.success
+        result = mash.run()
+        pungi_wrapper().compose_repo.assert_called_once()
+        assert mash.success
+        assert result == "Mash successfull"
+
+    @mock.patch("bodhi.server.consumers.masher.Compose")
+    @mock.patch("pungi.notifier.PungiNotifier")
+    @mock.patch("bodhi.server.consumers.masher.PungiWrapper")
+    def test_run_exception(self, pungi_wrapper, *args):
+        compose_id = "f27-modular-updates"
+        pungi_conf = mock.Mock()
+        target_dir = "/tmp"
+        variants_conf = mock.Mock()
+        logger = mock.Mock()
+        pungi_wrapper.init_compose_dir = mock.Mock()
+        mash = PungiMashThread(compose_id, target_dir, pungi_conf, variants_conf, logger)
+
+        assert not mash.success
+        pungi_wrapper().compose_repo.side_effect = Exception("Mash exception!")
+
+        with pytest.raises(Exception):
+            mash.run()
+            pungi_wrapper().compose_repo.assert_called_once()
+
+        assert not mash.success
+
+
+class TestPungiMasherThread(object):
+
+    def setup_method(self, method):
+        release = "f27-modular-updates"
+        request = "stable"
+        updates = []
+        agent = "mcurlej"
+        logger = mock.Mock()
+        db = mock.Mock()
```

Comment:
You don't need to mock the db - we have a nice BaseTest class that sets up a working DB which makes the tests more meaningful.